stix2viz/stix2viz/stix2viz.js

```diff
@@ -779,6 +800,64 @@ function makeEdgeObject(sourceRef, targetRef, label, stixId=null)
     return edge;
 }
 
+const timelineTimestamps = {                                   
+    "attack-pattern": ["modified", "created"],
+    "campaign": ["last_seen", "first_seen", "modified", "created"],
+    "course-of-action": ["modified", "created"],
+    "event": ["start_time", "modified", "created"],
+    "identity": ["modified", "created"],
+    "task": ["start_time", "modified", "created"],
+    "incident": ["modified", "created"],
+    "indicator": ["modified", "created"],
+    "infrastructure": ["modified", "created"],
+    "intrusion-set": ["last_seen", "first_seen", "modified", "created"],
+    "location": ["created"],
+    "malware": ["last_seen", "first_seen", "modified", "created"],
+    "malware-analysis": ["modified", "created"],
+    "note": ["modified", "created"],
+    "observed-data": ["last_observed", "first_observed", "modified", "created"],
+    "opinion": ["modified", "created"],
+    "report": ["modified", "created"],
+    "task": ["start_time", "modified", "created"],
+    "threat-actor": ["last_seen", "first_seen","modified", "created"],
+    "tool": ["modified", "created"],
+    "vulnerability": ["modified", "created"],
+
+    "relationship": ["start_time", "modified", "created"],
+    "sighting": ["last_seen", "first_seen", "modified", "created"]
+};
+
+function determineTimestampFromConfig(stixObject, stixType, config)
+{
+    if (config.has(stixType)) {
+        let typeConfig = config.get(stixType);
+        if (typeConfig.has("timestampList"))
+            return determineTimestamp(stixObject, typeConfig.get("timestampList"), null);
+    }
+    return null
+}
+
+function determineTimestamp(stixObject, timestampList)
+{
+    for (let prop of timestampList)
+    {
+        if (stixObject.has(prop))
+            return new Date(stixObject.get(prop)).valueOf()
+    }
+    return null
+}
+
+function determineTimestampForSCO(stixObject, observedDataNodes)
+{
+    let sco_id = stixObject.get("id")
+    for (let obsData of observedDataNodes)
+    {
+        let obj_refs = stixObject.get("object_refs")
+        if (!(obj_refs == null) &&  obj_refs.includes(sco_id))
+            return determineTimestamp(stixObject, timelineTimestamps["observed-data"])
```

Comment:
You are checking stixObject for observed-data properties, I think.  Maybe you meant to pass obsData here?

Reply:
good catch